### DISCUSSION TITLE
[BUG] Error when using the satellite imagery video

### DISCUSSION THREAD

**Discussion by PinRuei** [All4Gis/QGISFMV#116] [General]
**Describe the bug**
I want to try satellite motion video in QGISFMV, when I create the MISB using video multiplexer, I only can see the frame center, others cannot be shown (like footprint, beams, etc).
Because of small FOV (0.3), I change the form of FOV fill-in UI into float. Another issue is the big number of height(around 50,000 meters), I think this is the problem causing the error too.

**Expected behavior**
I think it can solve with the video which has the big number of height and small number of FOV. I had tried to check the code but I couldn't find where the code can control this problem(no footprint, platform, beams be shown).

**Screenshots**
Only can see the frame center
![image](https://user-images.githubusercontent.com/75609773/122893952-5bd32580-d379-11eb-8715-d7b043393a44.png)

**Desktop (please complete the following information):**
 - OS:  Windows 10
 - QGIS Version 3.18
 - Plugin Version 1.15

**Additional context**
If you can tell me where is the code of drawing footprint and beams, maybe I can try to fix it.
Thanks for the great job of providing QGISFMV.

**Comment by All4Gis** [comment]
Hi, yes, the code to create a MISB video is https://github.com/All4Gis/QGISFMV/blob/master/code/manager/QgsMultiplexor.py

To calculate and create the corners to generate the footprints is in https://github.com/All4Gis/QGISFMV/blob/master/code/utils/QgsFmvUtils.py#L966 or https://github.com/All4Gis/QGISFMV/blob/master/code/utils/QgsFmvUtils.py#L1043


and the function to update the layers, beams, footprints etc https://github.com/All4Gis/QGISFMV/blob/master/code/utils/QgsFmvUtils.py#L766

**Comment by PinRuei** [comment]
Thanks for the help.
How about the big number of height and small FOV?
Do you have any opinion on these two condition?

**Comment by All4Gis** [comment]
Nopre sorry, I don't have much time right now and I would have to study this case better.

**Comment by PinRuei** [comment]
Thanks, can I know how to change the packet that the video multiplexor created?
I want to add another in it, is this feasible?

![image](https://user-images.githubusercontent.com/75609773/123594163-f6bc7b80-d821-11eb-96fa-bcf636b5ab0a.png)

**Comment by All4Gis** [comment]
change the packet created by the multiplexer of this plugin, or change a packet of a multiplexed video, that part you would have to implement to extract the packets, modify them and multiplex again.But if you are using this multiplexor, debug the code and fix the error, maybe some calculation or data is not correct, check the code   https://github.com/All4Gis/QGISFMV/blob/master/code/manager/QgsMultiplexor.py

**Comment by PinRuei** [comment]
Do you know where the code can change the update rate from 2 second to 1 second?

**Comment by All4Gis** [comment]
depend of the pass time, https://github.com/All4Gis/QGISFMV/blob/ff1f105166c2fc6b0616ed468a484bde2c1a360d/code/player/QgsFmvPlayer.py#L1165

and the default value is
https://github.com/All4Gis/QGISFMV/blob/ff1f105166c2fc6b0616ed468a484bde2c1a360d/code/manager/QgsManager.py#L66


But, honestly I dont remenber well, maybe is this value `self.update_interval = 2000`, is used to update the player.

cheers

**Comment by PinRuei** [comment]
Hello, do you know where the code can change the svg icon of platform symbol?
I change it in https://github.com/All4Gis/QGISFMV/blob/master/code/manager/QgsFmvStyles.py #L37
and save the new svg icon in https://github.com/All4Gis/QGISFMV/tree/master/code/images/platforms
but it got the error.
